base/io.jl

```diff
@@ -1087,10 +1124,10 @@ julia> String(readavailable(buf))
 ```
 """
 function skipchars(predicate, io::IO; linecomment=nothing)
-    while !eof(io)
-        c = read(io, Char)
+    for c in readeach(io, Char)
         if c === linecomment
-            readline(io)
```

Comment:
`readline(io)` is not valid anymore in this context? (sorry if this ovious!)

Reply:
Ok, I see from a commit message that this might be to avoid allocation which presumably happens in `readline`? you might add a comment to clarify this and prevent future "simplifications" changing this back to "readline" :)